integrations/appsensor-integration-prometheus/src/main/java/org/owasp/appsensor/integration/prometheus/PrometheusEmitter.java

```diff
@@ -0,0 +1,187 @@
+package org.owasp.appsensor.integration.prometheus;
+
+import io.prometheus.client.exporter.PushGateway;
+import org.apache.commons.lang3.StringUtils;
+import org.owasp.appsensor.core.Attack;
+import org.owasp.appsensor.core.Event;
+import org.owasp.appsensor.core.Response;
+import org.owasp.appsensor.core.listener.SystemListener;
+import org.owasp.appsensor.core.storage.ResponseStoreListener;
+import org.owasp.appsensor.integration.prometheus.metrics.AbstractMetrics;
+import org.owasp.appsensor.integration.prometheus.metrics.AttackMetrics;
+import org.owasp.appsensor.integration.prometheus.metrics.EventMetrics;
+import org.owasp.appsensor.integration.prometheus.metrics.ResponseMetrics;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.env.Environment;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.util.ArrayList;
+import java.util.Collection;
+
+/**
+ * This is the Prometheus Pushgateway Emitter.
+ * <p>
+ * It is notified whenever new {@link Event}, {@link Attack}
+ * or {@link Response} objects are added to the system.
+ * <p>
+ * The implementation sends events/attacks/responses metrics to
+ * pushgateway every time they are received.
+ * <p>
+ * <p>Note: This class requires certain settings to run properly. This/these can be set as
+ * environment variables ('export my_var="some_value"') or environment
+ * properties ('-Dmy_var=some_value') set at the JVM</p>
+ * <ul>
+ * <li><em>PROMETHEUS_PUSHGATEWAY_ADDRESS</em> - the address used to the Prometheus pushgateway, e.g. "127.0.0.1:9091/"</li>
+ * </ul>
+ *
+ * @author Robert Przystasz  (robert.przystasz@gmail.com)
+ * @author Bartosz Wyględacz (bartosz.wygledacz@gmail.com)
+ * @author Michal Warzecha   (mwarzechaa@gmail.com)
+ * @author Magdalena Idzik   (maddie@pwnag3.net)
+ * @since 2.2.1
+ */
+
+@Named
+@ResponseStoreListener
```

Comment:
@sidewalker @wylly just curious - what did you need to include the ResponseStoreListener annotation for? With the SystemListener, it shouldn't be necessary.